go/multipooler/manager/rpc_initialization.go

```diff
@@ -0,0 +1,360 @@
+		return nil, mterrors.New(mtrpcpb.Code_UNAVAILABLE, "pgctld client not available")
+	}
+
+	restartReq := &pgctldpb.RestartRequest{
+		AsStandby: true,
+		Mode:      "fast",
+	}
+	if _, err := pm.pgctldClient.Restart(ctx, restartReq); err != nil {
+		return nil, mterrors.Wrap(err, "failed to restart PostgreSQL as standby")
+	}
+
+	// 5. Wait for database connection
+	if err := pm.waitForDatabaseConnection(ctx); err != nil {
+		return nil, mterrors.Wrap(err, "failed to connect to database")
+	}
+
+	// 6. Set consensus term
+	if pm.consensusState != nil {
+		if err := pm.consensusState.UpdateTermAndSave(ctx, req.ConsensusTerm); err != nil {
+			return nil, mterrors.Wrap(err, "failed to set consensus term")
+		}
+	}
+
+	pm.logger.InfoContext(ctx, "Successfully initialized pooler as standby", "shard", pm.getShardID(), "term", req.ConsensusTerm)
+	return &multipoolermanagerdatapb.InitializeAsStandbyResponse{
+		Success:  true,
+		FinalLsn: finalLSN,
+	}, nil
+}
+
+// InitializationStatus returns the initialization status of this pooler
+// Used by multiorch coordinator to determine what initialization scenario to use
+func (pm *MultiPoolerManager) InitializationStatus(ctx context.Context, req *multipoolermanagerdatapb.InitializationStatusRequest) (*multipoolermanagerdatapb.InitializationStatusResponse, error) {
+	pm.logger.DebugContext(ctx, "InitializationStatus called", "shard", pm.getShardID())
+
+	// Get WAL position (ignore errors, just return empty string)
+	walPosition, _ := pm.getWALPosition(ctx)
+
+	resp := &multipoolermanagerdatapb.InitializationStatusResponse{
+		IsInitialized:    pm.isInitialized(),
+		HasDataDirectory: pm.hasDataDirectory(),
+		PostgresRunning:  pm.isPostgresRunning(ctx),
+		Role:             pm.getRole(ctx),
+		WalPosition:      walPosition,
+		ShardId:          pm.getShardID(),
+	}
+
+	// Get consensus term if available
+	if pm.consensusState != nil {
+		term, err := pm.consensusState.GetCurrentTermNumber(ctx)
+		if err == nil {
+			resp.ConsensusTerm = term
+		}
+	}
+
+	return resp, nil
+}
+
+// Helper methods
+
+// isInitialized checks if the pooler has been initialized (has data directory and multigres schema)
+func (pm *MultiPoolerManager) isInitialized() bool {
+	if !pm.hasDataDirectory() {
+		return false
+	}
+
+	if pm.db == nil {
+		return false
+	}
+
+	// Check if multigres schema exists
+	exists, err := pm.querySchemaExists(context.Background())
+	return err == nil && exists
+}
+
+// hasDataDirectory checks if the PostgreSQL data directory exists
+func (pm *MultiPoolerManager) hasDataDirectory() bool {
+	if pm.config == nil || pm.config.PoolerDir == "" {
+		return false
+	}
+
+	dataDir := filepath.Join(pm.config.PoolerDir, "pg_data")
+	info, err := os.Stat(dataDir)
+	if err != nil {
+		return false
+	}
+
+	return info.IsDir()
+}
+
+// isPostgresRunning checks if PostgreSQL is currently running
+func (pm *MultiPoolerManager) isPostgresRunning(ctx context.Context) bool {
+	if pm.pgctldClient == nil {
```

Comment:
small nit - I wonder, if we should remove this check. Seems like calling Status should be the canonical way of checking this

Reply:
It's a standard nil check to avoid a possible nil dereference on line 242. It doesn't hurt to leave it in.
If you mean the whole function is redundant, and should be replaced by a call to some existing function, we can do that as a follow up.